src/lib/hooks/useAppState.js

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+import { AppState } from 'react-native'
+
+export default settings => {
+  const { onChange, onForeground, onBackground } = settings || {}
+  const [appState, setAppState] = useState(AppState.currentState)
+
+  useEffect(() => {
+    const handleAppStateChange = nextAppState => {
+      if (appState !== nextAppState) {
+        const cb = nextAppState === 'active' ? onForeground : onBackground
+        cb()
+        onChange(nextAppState)
+      }
+      setAppState(nextAppState)
+    }
+    AppState.addEventListener('change', handleAppStateChange)
+
+    return () => AppState.removeEventListener('change', handleAppStateChange)
+  }, [appState, settings])
```

Comment:
and here you could listen to the passed callbacks functions changes only (as you're referencing then only in your effect)

e.g. `[appState, onChange, onForeground, onBackground]`

Reply:
i could but that would be too verbose